### PR TITLE
Make SciView.addVolume accept RAI instead of II

### PR DESCRIPTION
**Pull request by tpietzsch**
Addresses #144

### CODE REVIEW

File: src/main/java/sc/iview/SciView.java [hunk 1]
```diff
@@ -1101,48 +1101,44 @@ public void setColormap( Node n, ColorTable colorTable ) {
...
+            ushorts.forEach( t -> byteBuffer.putShort( ( t.getShort() ) ) );
+        } else { // if( type instanceof FloatType )
+            @SuppressWarnings("unchecked") final IterableInterval< FloatType > floats = ( IterableInterval< FloatType > ) Views.flatIterable( image );
+            floats.forEach( t -> byteBuffer.putFloat( t.get() ) );
         }
         byteBuffer.flip();
 
         v.readFromBuffer( name, byteBuffer, dimensions[0], dimensions[1], dimensions[2], voxelDimensions[0],
-                          voxelDimensions[1], voxelDimensions[2], nType, bytesPerVoxel );
+                          voxelDimensions[1], voxelDimensions[2], nType, ( int ) nType.getSizeInBytes() );
```
Comment (ctrueden) [inline src/main/java/sc/iview/SciView.java]: Should we assert that `nType.getSizeInBytes() <= Integer.MAX_VALUE` above, just to be safe?

File: src/main/java/sc/iview/SciView.java [hunk 2]
```diff
@@ -1101,48 +1101,44 @@ public void setColormap( Node n, ColorTable colorTable ) {
...
-            cursor.fwd();
-            if( voxelType == UnsignedByteType.class ) {
-                byteBuffer.put( ( byte ) ( ( ( UnsignedByteType ) cursor.get() ).get() ) );
-            } else if( voxelType == UnsignedShortType.class ) {
-                byteBuffer.putShort( ( short ) Math.abs( ( ( UnsignedShortType ) cursor.get() ).getShort() ) );
-            } else if( voxelType == FloatType.class ) {
-                byteBuffer.putFloat( ( ( FloatType ) cursor.get() ).get() );
-            }
+        final ByteBuffer byteBuffer = ByteBuffer.allocateDirect(
+                ( int ) ( nType.getSizeInBytes() * Intervals.numElements( dimensions ) ) );
```
Comment (ctrueden) [inline src/main/java/sc/iview/SciView.java]: There is `org.scijava.util.ArrayUtils.safeMultiply32` to do a product of longs and guarantee the result fits into an `int`—or else an exception is thrown. Should we use it here?

File: src/main/java/sc/iview/SciView.java (hunk 1 above)
Comment (tpietzsch) [reply src/main/java/sc/iview/SciView.java]: No. It's an enum and the maximum that we can get is 8.

File: src/main/java/sc/iview/SciView.java (hunk 2 above)
Comment (tpietzsch) [reply src/main/java/sc/iview/SciView.java]: Sure, I'll add it.

**Comment by skalarproduktraum** [comment]
@kephale Shall we merge this?

**Pull request merged**